benchmark/single-source/IntegerParsing.swift

```diff
@@ -0,0 +1,85 @@
+//===--- IntegerParsing.swift -------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import TestsUtils
+
+public let IntegerParsing = [
+  BenchmarkInfo(name: "ParseInt.Decimal",
+    runFunction: run_ParseIntFromDecimal,
+    tags: [.validation, .api],
+    setUpFunction: { blackHole(decimalStrings) }),
+  BenchmarkInfo(name: "ParseInt.Binary",
+    runFunction: run_ParseIntFromBinary,
+    tags: [.validation, .api],
+    setUpFunction: { blackHole(binaryStrings) }),
+  BenchmarkInfo(name: "ParseInt.Hex",
+    runFunction: run_ParseIntFromHex,
+    tags: [.validation, .api],
+    setUpFunction: { blackHole(hexStrings) }),
+  BenchmarkInfo(name: "ParseInt.UncommonRadix",
+    runFunction: run_ParseIntFromUncommonRadix,
+    tags: [.validation, .api],
+    setUpFunction: { blackHole(uncommonRadixStrings) }),
+]
+
+private let values: [Int] = Array(-1000...1000) // Give extra weight to low ints
+  + (0..<2000).map { _ in Int.random(in: .min ... .max) }
```

Comment:
It may be worth actually splitting these into two separate tests for each radix.